app/domain/authentication/authn_jwt/authenticator.rb

```diff
@@ -0,0 +1,140 @@
+require 'command_class'
+
+module Authentication
+  module AuthnJwt
+    # Generic JWT authenticator that receive JWT vendor configuration and uses to validate that the authentication
+    # request is valid, and return conjur authn token accordingly
+    Authenticator = CommandClass.new(
+      dependencies: {
+        token_factory: TokenFactory.new,
+        logger: Rails.logger,
+        audit_log: ::Audit.logger,
+        validate_origin: ::Authentication::ValidateOrigin.new,
+        role_class: ::Role,
+        webservice_class: ::Authentication::Webservice,
+        validate_role_can_access_webservice: ::Authentication::Security::ValidateRoleCanAccessWebservice.new,
+        role_id_class: Audit::Event::Authn::RoleId
+      },
+      inputs: %i[jwt_configuration authenticator_input]
+    ) do
+      extend(Forwardable)
+      def_delegators(:@authenticator_input, :account, :username, :client_ip, :authenticator_name, :service_id)
+
+      def call
+        validate_and_decode_token
+        get_jwt_identity
+        validate_user_has_access_to_webservice
+        validate_origin
+        validate_restrictions
+        audit_success
+        new_token
+      rescue => e
+        audit_failure(e)
+        raise e
+      end
+
+      private
+
+      def validate_and_decode_token
+        @logger.debug(LogMessages::Authentication::AuthnJwt::CallingValidateAndDecodeToken.new)
+        @jwt_configuration.validate_and_decode_token
+        @logger.debug(LogMessages::Authentication::AuthnJwt::ValidateAndDecodeTokenPassed.new)
+      end
+
+      def get_jwt_identity
+        @logger.debug(LogMessages::Authentication::AuthnJwt::CallingGetJwtIdentity.new)
+        jwt_identity
+        @logger.debug(LogMessages::Authentication::AuthnJwt::GetJwtIdentityPassed.new)
+        @logger.info(LogMessages::Authentication::AuthnJwt::FoundJwtIdentity.new(jwt_identity))
+        @jwt_identity_initialized = true
```

Comment:
this should not be done. please use memoization instead. instead of later asking `if @jwt_identity_initialized` you should ask `if jwt_identity `

Reply:
Used defined?(@jwt_identity)